src/eradiate/radprops/rayleigh.py

```diff
@@ -22,13 +22,46 @@
 class _BATES_1984_DATA(metaclass=Singleton):
     # Lazy loader for Bates (1984) King correction data
     def __init__(self):
-        self._data = None
+        self._king_factor = None
+        self._wavelength = None
 
     @property
-    def data(self):
-        if self._data is None:
-            self._data = data.load_dataset("spectra/optics/bates_1984.nc")
-        return self._data
+    def king_factor(self):
+        if self._king_factor is None:
+            self._load_dataset()
+        return self._king_factor
+
+    @property
+    def wavelength(self):
+        if self._wavelength is None:
+            self._load_dataset()
+        return self._wavelength
+
+    def interp(self, w: pint.Quantity = ureg.Quantity(0.550, "micron")):
```

Comment:
Since all fields are unitless, I think we can move the conversion to the upper level.
```suggestion
    def interp(self, w: float = 0.550):
```

Reply:
Are there any cases where `w` can be an array?

Reply:
That is a good point, I overlooked that. Indeed, `w` should accept an array-like.